cmd/containerd-shim-runhcs-v1/service_internal.go

```diff
@@ -215,7 +216,14 @@ func (s *service) deleteInternal(ctx context.Context, req *task.DeleteRequest) (
 	if err != nil {
 		return nil, err
 	}
-	// TODO: We should be removing the task after this right?
+
+	if req.ExecID == "" {
+		// We've successfully deleted the init exec. Compute resources
+		// should be closed, layers umounted and resources released.
+		// This task should now be done.
+		s.taskOrPod = atomic.Value{}
```

Comment:
This is not exactly correct. Ideally we should store an empty value of the already stored type (&hcsTask{} or &pod{}). The problem is that the tests will use a ```testShimPod{}``` or a ```testShimTask{}```. The ```Store()``` and ```Swap()``` functions will check the underlying type of the new value we want to store, and panic if it differs from the already stored type. A value of ```nil``` is not allowed.

Reply:
Wrote a helper variable that is overwritten during tests, which returns the proper type.